appserver/tests/payara-samples/samples/microprofile-config-alias-properties/src/test/java/fish/payara/samples/microprofile/config/alias/MicroProfileConfigAliasTest.java

```diff
@@ -0,0 +1,105 @@
+import com.gargoylesoftware.htmlunit.WebClient;
+import fish.payara.samples.CliCommands;
+import fish.payara.samples.NotMicroCompatible;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.net.URL;
+import java.nio.file.Paths;
+
+/**
+ * @author Andrew Pielage <andrew.pielage@payara.fish>
+ */
+@RunWith(Arquillian.class)
+@NotMicroCompatible
+public class MicroProfileConfigAliasTest {
+
+    @ArquillianResource
+    private URL url;
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class, "microprofile-config-alias.war")
+                .addPackage("fish.payara.samples.microprofile.config.alias")
+                .addAsManifestResource(new File("src/main/resources/META-INF/microprofile-config-alias.properties"));
+    }
+
+    @BeforeClass
+    public static void createPasswordAlias() {
+        CliCommands.payaraGlassFish("create-password-alias", "-W", Paths.get("").toAbsolutePath() + "/src/test/resources/passwordfile.txt", "wibbles");
+
+        // Deployment actually happens before @BeforeClass, and the PasswordAlias config source requires an application refresh to update, so disable and enable it
+        CliCommands.payaraGlassFish("set", "servers.server.server.application-ref.microprofile-config-alias.enabled=false");
+        CliCommands.payaraGlassFish("set", "servers.server.server.application-ref.microprofile-config-alias.enabled=true");
+    }
+
+    @AfterClass
+    public static void deletePasswordAlias() {
+        CliCommands.payaraGlassFish("delete-password-alias", "wibbles");
+    }
+
+    @Test
+    public void testAliasSubstitution() throws Exception {
+        TextPage page = new WebClient().getPage(url + "ConfigServlet");
+        System.out.println(page.getContent());
+
+        Assert.assertTrue(page.getContent().contains("Normal Notation: wobbles"));
+        Assert.assertTrue(page.getContent().contains("Substitution Notation: wobbles"));
+        Assert.assertTrue(page.getContent().contains("Password Alias from File: wobbles"));
+        Assert.assertTrue(page.getContent().contains("System Property Alias from File: Tiddles!"));
+        Assert.assertTrue(page.getContent().contains("System Property Alias from File: Tiddles!"));
+        Assert.assertTrue(page.getContent().contains("Environment Variable Alias referencing System Property Alias from File: Dobbles"));
+        Assert.assertTrue(page.getContent().contains("Environment Variable Alias and System Property Alias from File (same property): Bibbles and Bobbles"));
+    }
```

Comment:
I'd like to see a message in connection with `assertTrue` and `contains`. Maybe its a good idea to create a little helper method that in case of failure has a message like `"Expected contains: Normal Notation: wobbles"` and so forth.